Cubical/HITs/Hopf.agda

```diff
@@ -1,33 +1,387 @@
-{-# OPTIONS --cubical --safe #-}
+{-# OPTIONS --cubical #-}
 module Cubical.HITs.Hopf where
 
 open import Cubical.Core.Primitives
 open import Cubical.Core.Prelude
 open import Cubical.Core.Glue
 
+open import Cubical.Foundations.Equiv
+
+open import Cubical.Data.Int
+open import Cubical.Data.Prod
+
 open import Cubical.HITs.S1
 open import Cubical.HITs.S2
+open import Cubical.HITs.S3
 open import Cubical.HITs.Susp
+open import Cubical.HITs.Join
+
+data Interval : Set where
+  I0 : Interval
+  I1 : Interval
+  seg : I0 ≡ I1
+
+PseudoHopf : Set
+PseudoHopf = (S¹ × Interval) × S¹
+
+PseudoHopf-π1 : PseudoHopf → S¹
+PseudoHopf-π1 ((y , _) , _) = y
+
+PseudoHopf-π2 : PseudoHopf → S¹
+PseudoHopf-π2 (_ , x) = x
+
+Border : (x : S¹) → (j : I) → Partial (j ∨ (~ j)) (Σ Set (λ T → T ≃ S¹))
+Border x j (j = i0) = S¹ , rot x , rotIsEquiv x
+Border x j (j = i1) = S¹ , idEquiv S¹
+
+Border1 : (x : S¹) → (j : I) → Partial (j ∨ (~ j)) Set
+Border1 x j y = Border x j y .fst
+
+Border2 : (x : S¹) → (j : I) → PartialP (j ∨ (~ j)) (λ y → Border1 x j y ≃ S¹)
+Border2 x j y = Border x j y .snd
 
--- Hopf fibration using S²
-HopfS² : S² → Set
-HopfS² base = S¹
-HopfS² (surf i j) = Glue S¹ (λ { (i = i0) → _ , idEquiv S¹
-                               ; (i = i1) → _ , idEquiv S¹
-                               ; (j = i0) → _ , idEquiv S¹
-                               ; (j = i1) → _ , _ , rotIsEquiv (loop i) } )
-
--- Hopf fibration using more direct definition of the rot equivalence
-HopfS²' : S² → Set
-HopfS²' base = S¹
-HopfS²' (surf i j) = Glue S¹ (λ { (i = i0) → _ , rotLoopEquiv i0
-                                ; (i = i1) → _ , rotLoopEquiv i0
-                                ; (j = i0) → _ , rotLoopEquiv i0
-                                ; (j = i1) → _ , rotLoopEquiv i } )
-
--- Hopf fibration using suspension of S¹
 HopfSuspS¹ : SuspS¹ → Set
 HopfSuspS¹ north = S¹
 HopfSuspS¹ south = S¹
-HopfSuspS¹ (merid x i) = ua (_ , rotIsEquiv x) i
+HopfSuspS¹ (merid x j) = Glue S¹ (Border x j)
+
+-- Total space of the fibration
+TotalSpace : Set
+TotalSpace = Σ SuspS¹ HopfSuspS¹
+
+unglueb : (x : S¹) → (j : I) → Glue S¹ (Border x j) → S¹
+unglueb x j y = unglue {_} _ {Border1 x j} {Border2 x j} y
+
+-- Forward direction
+filler-1 : I → (j : I) → (y : S¹) → Glue S¹ (Border y j) → join S¹ S¹
+filler-1 i j y x = hfill (λ t → λ { (j = i0) → inl (rotInv-1 x y t)
+                                  ; (j = i1) → inr x })
+                         (inc (push (rot (unglueb y j x) (flip y)) (unglueb y j x) j)) i
+
+Hopf→Join : TotalSpace → join S¹ S¹
+Hopf→Join (north , x) = inl x
+Hopf→Join (south , x) = inr x
+Hopf→Join (merid y j , x) = filler-1 i1 j y x
```

Comment:
This seems simpler than

https://github.com/mortberg/cubicaltt/blob/pi4s3_dimclosures/examples/brunerie.ctt#L688

which is what I used in my previous attempts to compute the Brunerie number using cubicaltt (the definition of the t map comes from page 150 of Guillaume's thesis)